build.rs

```diff
@@ -692,9 +692,9 @@ fn perlasm_src_dsts(out_dir: &Path, asm_target: &AsmTarget) -> Vec<(PathBuf, Pat
     {
         // Appease the borrow checker.
         let mut maybe_synthesize = |concrete, synthesized| {
-            let concrete_path = PathBuf::from(concrete);
+            let concrete_path = absolute_path_from_manifest_dir_relative_path(concrete);
             if srcs.contains(&concrete_path) {
-                let synthesized_path = PathBuf::from(synthesized);
+                let synthesized_path = absolute_path_from_manifest_dir_relative_path(synthesized);
```

Comment:
I think we don't need to make these absolute here. Instead, we should make the paths absolute just before we pass them to cc-rs and/or `perlasm()`.

Reply:
Makes sense - done